### ISSUE TITLE
[Housekeeping] DRY CI with Action Composition

### ISSUE THREAD

**Issue by MadsBuchmann** [kirbydesign/designsystem#1518]
**Short description of housekeeping request**

I would like to investigate if we can [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) our GitHub Actions workflows (located in `.github/workflows`) by utilising composite run steps and actions: 
- https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/ 
- https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/

We have quite a lot of repeating steps. For example, if we are to change the node version used, we have to change it 3 - 5 different places.  

This makes it easy to make mistakes and makes it hard to get an overview. 

### Kick Off

* [ ] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] Implement/update unit tests
* [ ] Create a draft implementation and push to Github
* [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request

### Code Review

* [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

**Comment by MadsBuchmann** [comment]
I am marking this as blocked - and leaving a little writeup for whoever will take on this issue in the future 😄 

# Why block it?
When starting this issue I tought it would be a problem storing composite actions in the same repo and not having to put them in an external one. This was quite easy actually as Github supports local composite actions: 
```
    - name: Checkout repo    
      uses: actions/checkout@v2    
    - name: Say Hello                                                                                                                                                                                 
      uses: ./.github/actions/say-hello    
      with:        
        name: Mads   
```

It unfortunately turns out that you can't have `uses` steps in your composite actions. In other words, you are not able to use other Github Actions such as: `actions/cache@v2` or `actions/checkout@v2` etc. 

It seems like It is a work in progress and something Github want to do but it is not ready yet. We should therefore leave this blocked until that feature is ready. 

For more see the following resources: 
- https://github.com/actions/runner/issues/438
- https://github.com/actions/runner/issues/646

# An Alternative solution! 
An alternative solution I've stumbled upon when working with this - is to implement YAML anchors via a build script that can be executed pre-commit. I found it here and the author also links to an example repo: https://github.com/akctions/runner/issues/438#issuecomment-722778085 

I played around with it for maybe 30 minutes but had some problems with my node version. I decided to not use anymore time on it as I do not know how this alternative solution will be received 😄

# Ideas for Composite Actions
I've located two routines that are being executed several times in our workflows that I would like to turn into reusable composite actions. Unfortunately both of these uses other Github Actions and can therefore not be put into a composite run step.

The first I would like to put into a 'Setup Kirby environment' action:
```
      - name: Install Node.js    
        uses: actions/setup-node@v2    
        with:    
          node-version: ${{ env.NODE_VERSION }}    
      - name: Fetch Node Modules Cache    
        uses: actions/cache@v2    
        id: cache    
        with:    
          path: |     
            **/node_modules    
          key: node-modules-${{ runner.os }}-${{env.NODE_VERSION}}-${{hashFiles('**/package.json')}}-${{hashFiles('**/package-lock.json')}}    
      - name: Clean Install NPM Dependencies    
        if: steps.cache.outputs.cache-hit != 'true'    
        run: npm ci    
      - name: Run Postinstall Script    
        if: steps.cache.outputs.cache-hit == 'true'    
        run: npm run postinstall     
```

The above is executed around 5 times in our workflows at the time of writing. 


Secondly there are 4 places in the code where  _very_ similar steps are executed, these are: 
* "Set deployment status to in progress" 
* "Set deployment status to failed" 
* "Set deployment status to success" 
* "Set deployment status to inactive". 

Therefore  I would like to create a "Set deployment status"  action which should take a status as input.

**Comment by MadsBuchmann** [comment]
In honor of the backlog pruning being done by @alxzak I'm closing this issue. We can get back to it whenever it is relevant. 

More info: [link](https://github.com/kirbydesign/designsystem/discussions/1704)

**Comment by MadsBuchmann** [comment]
I think this is still relevant in order to simplify the pipeline. Go away stalebot.

**Comment by MadsBuchmann** [comment]
### A suggestion from the review of #1402 that could be carried out as part of this issue. 

[Original comment](https://github.com/kirbydesign/designsystem/pull/1402#discussion_r736272054) from PR: 
> Now that GitHub supports action composition we could consider plopping line 20 to 46 into some sort of "setup" action. Then we could also replace it in the other files.
>
>Might not be related to this issue... Buuuuut we do have the file open anyways 👀
>
>Alternatively I can open a new issue and fix it there, which can then be merged into this branch.

Related code: `.github/workflows/publish.yml`
